addons/product/models/product_template.py

```diff
@@ -1189,6 +1190,16 @@ def _cartesian_product(self, product_template_attribute_values_per_line, parent_
         while True:
             current_line_values = product_template_attribute_values_per_line[line_index]
             current_ptav_index = value_index_per_line[line_index]
+
+            # For multi-checkbox attribute, the list is empty as we want to start without any selected value
+            if not current_line_values:
+                if line_index == len(product_template_attribute_values_per_line) - 1:
+                    # submit combination if we're on the last line
+                    yield partial_combination
+                else:
+                    line_index += 1
+                    continue
```

Comment:
I don't see any check here to make sure we're in a multi-checkbox case

Reply:
`current_line_values` is a list of available ptavs, but is empty (only) for 'multi' (see https://github.com/odoo/odoo/pull/135174/files#diff-921ec733c3821301f62cd0d96e45990e7636f5db56a10ae76aed009e380bf149R1284)